chrisdev_transportExtension_1203750651534223

chriscodev opened the pull request.

darlp5589 commented:
Got this. Will do.

triveni-codev commented:
Hi @darlp5589 , The commit operation will now communicate well with the backend [by doing this finding by tian](https://triveni-codev.slack.com/archives/C01UH4NQ49Z/p1674183170212219?thread_ts=1674175509.404849&cid=C01UH4NQ49Z)

Thanks.

darlp5589 commented:
This is noted.

darlp5589 reviewed (changes requested):
Done checking PR. Here are my concerns:

1. i1: Should be Checkmark instead of Xmark in the Ok button.
2. i2: Should be Add text instead of Ok with a Checkmark before it.
3. i3: This will show up if I click cancel button on "Add Bonded BSIDs" modal then proceed to add again Bonded BSIDs.
4. i4: Upon comparing this with client app, it should accept a value between from 0 to 65535. This accepts beyond 65535.
5. i5: Tried to update what I just added and cleared the bonded bsid value until that checkmark on the right turned black then clicked cancel button. When I update again, that checkmark remains black even if it contains the original value.
6. i6: This shows up when I added a transport extension (only on dirty change / no big commit) and attempted to update the newly added transport.
7. i7: If I procced and clicked OK on the concern #6 above, that console error on the right will show up if I try to update the extension so I believe they are related.
8. i8: I have an existing ATSC3 type Transport and added extension on it the clicked Update. When I try to update it again, the newly added extension is gone.
9. i9: I can add letters through copy-paste.
10. I added an extension the remove it but clicked Prev button then proceed to extension and noticed that the removed extension is still there.
11. In relation to concern #10, after clicking removed extension then clicked Add then proceed on updating the said transport and go to extension, it's still there.
12. i10: For good measure, I still received an error when I added a transport with an extension then clicked Big Commit.



i1: 
![image](https://user-images.githubusercontent.com/101298313/213618176-ef56fbf7-27b7-41ff-b550-15209d8827a8.png)


i2: 
![image](https://user-images.githubusercontent.com/101298313/213618010-751cbbb1-42a2-43dc-bd9e-60f0953df33a.png)

i3: 
![image](https://user-images.githubusercontent.com/101298313/213618347-e68cba64-4191-449d-883f-3ace623a8165.png)

i4: 
![image](https://user-images.githubusercontent.com/101298313/213618792-012cdbcf-e251-46e2-9806-134acd124ba1.png)

i5: 
![image](https://user-images.githubusercontent.com/101298313/213619051-73287783-6454-4986-b6db-064a9243131a.png)

i6: 
![image](https://user-images.githubusercontent.com/101298313/213619733-072d975b-0c56-4764-aba7-b42d08061f83.png)

i7: 
![image](https://user-images.githubusercontent.com/101298313/213620117-9eaec29f-f5f0-418e-b472-249d6f890231.png)

i8: 
![image](https://user-images.githubusercontent.com/101298313/213620278-252cc176-fa18-427f-8fba-74b3e4a79061.png)

i9: 
![image](https://user-images.githubusercontent.com/101298313/213620858-6f5b240d-e9f2-42f3-bb9e-9a088384722a.png)

i10: 
![image](https://user-images.githubusercontent.com/101298313/213621893-664610c9-1626-4e90-9324-8158c05b2c0d.png)

chriscodev commented:
hi @darlp5589 , 

i have new update on this PR,  you  may now verify in your  end.

i1: Should be Checkmark instead of Xmark in the Ok button.
[DONE]

i2: Should be Add text instead of Ok with a Checkmark before it.
[DONE]

i3: This will show up if I click cancel button on "Add Bonded BSIDs" modal then proceed to add again Bonded BSIDs.
[DONE]

i4: Upon comparing this with client app, it should accept a value between from 0 to 65535. This accepts beyond 65535.
[DONE]

i5: Tried to update what I just added and cleared the bonded bsid value until that checkmark on the right turned black then clicked cancel button. When I update again, that checkmark remains black even if it contains the original value.
[DONE]

i6: This shows up when I added a transport extension (only on dirty change / no big commit) and attempted to update the newly added transport.
[DONE]

i7: If I procced and clicked OK on the concern #6 above, that console error on the right will show up if I try to update the extension so I believe they are related.
[DONE]

i8: I have an existing ATSC3 type Transport and added extension on it the clicked Update. When I try to update it again, the newly added extension is gone.
[DONE]

i9: I can add letters through copy-paste.

I added an extension the remove it but clicked Prev button then proceed to extension and noticed that the removed extension is still there.

In relation to concern #10, after clicking removed extension then clicked Add then proceed on updating the said transport and go to extension, it's still there.
[DONE]

i10: For good measure, I still received an error when I added a transport with an extension then clicked Big Commit.
[ this will be in the next testing, as we  have concern on api update
needs to be updated in svn ]

Cheers,
Christian

darlp5589 reviewed (changes requested):
Done checking fixes. Here are my concerns:

1. i1: It's still showing up once click cancel then proceed to add again.
2. Not related to Extensions but for good measure, as I did my set of testing fixes, I realized adding a transport with service groups and clicked Add then proceed on updating the service group name but clicked Cancel - it will still save the updated service group name even if cancelled.
3. Pending on Big Commit testing because of the concern on API update.


i1: 
![image](https://user-images.githubusercontent.com/101298313/213963311-9649f577-b9c5-4687-9813-d352455c7675.png)

darlp5589 reviewed (changes requested):
Done checking PR fixes. I still have concerns on the group settings name being updated and cancelled but still updating the value. See video below.

v1: https://www.loom.com/share/f957a3b5faea49d58244f0852331ca91

chriscodev commented:
hi @darlp5589 , 

this PR is for transport extension only, the fix for this is in this PR - https://github.com/triveni-codev/GBWebApp-dev/pull/514,


Cheers,
Christian

darlp5589 commented:
@chriscodev  Based on convo, this was already fixed on the previous PR and this PR wasn't branched out from the previous one. Should be good now except for the pending API request concern for big commit.
